### DISCUSSION TITLE
Can't install Synology SSL VPN Client

### DISCUSSION THREAD

**Discussion by ghost** [HotCakeX/Harden-Windows-Security#57] [Q&A] [answered]
I have been trying for days to find out, without success, which part of the script is blocking the installation of the Synology SSL VPN Client.
 script blocks the installation of Synology SSL VPN Client. It just doesn't happen. No popup or anything. 

Any ideas?

https://global.synologydownload.com/download/Utility/VPNPlusClient/1.4.6-0685/Windows/x86_64/Synology%20SSL%20VPN%20Client-1.4.6-0685_X64.msi?

**Comment by HotCakeX** [comment]
Hi, I just tried your installer and it installed successfully on my computer.

I'm using latest version of Windows 11 and applied all categories of the hardening script except for the Top security section and Smart App Control. Have you applied those 2 categories ? 

Are you using any 3rd party AV or similar software?

**Reply by ghost** [reply]
Hi HotCakeX,

Thanks for your quick response.

This has failed on two computers. On a virtual machine, which in the end I had to replace it. And on my laptop on which the script has been installed for weeks with apparently no other problems.

Regarding AV I use Kaspersky. But I have already tried to disable it and everything remains the same. And in the new virtual machine with Kaspersky and without the script, software works.
 
About those categories, I'm sure I have not applied "Top Security" but I am not so sure about SmartAPP.

**Reply by HotCakeX** [reply]
3rd party AVs are not compatible, I've mentioned it in another issue too but I will add it to the readme as well in the requirements section to make it more clear.

Microsoft Defender category and others are incompatible with 3rd party AVs. MDAV is much better than them and renders them redundant.

You can verify Smart App Control's status in Windows Security => App & browser control.

My suggestion: uninstall any 3rd party security solution and then install your software because the script has no problem with it.

**Reply by ghost** [reply]
FYI:

I uninstalled the antivirus, rebooted and... nothing.
With Kaspersky uninstalled, I reapplied the script and still can't instal the software.

If I press Shift: run as another user, the user and password context menu does not appear. It only happens with this package.

In the TEMP folder of the user profile, every time I run the installer, files of the following type are created: qtsingleapp-launch-dcb-1-lockfile

They are always "lockfile".

I'm sure it's something in the script, don't get me wrong, I don't want to blame you for anything, but this package (Synology) when you download it with Edge, it warns that this package is not signed.

If I go back to normal without the script, with my usual AV, everything works. But I want your script, I'm sysadmin and I think you've done an outstanding job!

**Reply by HotCakeX** [reply]
Thank you ^^
I understand but I'm not sure what else I can do on my side...

I tried the installer on both my system and a VM and it worked. Everything the script does is fully documented on the readme, there are only 2 things that i can think of that can prevent a program from installing.

One of them is Smart App Control and the other is Mandatory ASLR. Smart App Control wasn't on in my tests because I use my own WDAC policy but Mandatory ASLR was on in both of my tests and didn't prevent the installation. Only very few (3 that I know of) software has problem with Mandatory ASLR and it's because they're poorly designed.

The problem might be deeper and need a clean reinstall of Windows or a reset from the settings using cloud download option.

As far as I can tell the installer is fine but you're right, it's not signed

![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/ee2d6784-88b9-47b9-9a8a-c399a45c0678)

No icon was added to the desktop though, had to find it in programs folder.
`C:\Program Files\Synology\VPN Plus Client`

**Reply by ghost** [reply, marked as answer]
OMG I just discovered the problem! And it's something I'm sure many of us have seen hundreds of times.

The package was locked. Right click, unlock and it works!

Why did it lock it twice on both machines? Maybe Kaspersky + Windows Defeder (with your policies?) Don't know but it's clear it's already working.

Thank you very much for your help. It was something so obvious that I didn't even think of it.

**Reply by HotCakeX** [reply]
Oh I'm glad to know that 😊
It is very much possible for the 2 AVs to cause conflicts and result in things like that

Anytime ^^
